### PR TITLE
Added option to configure Lean server environment PATH

### PR DESCRIPTION
**Pull request by tydeu**
Addresses #22, enabling users to add paths to PATH environment variable used for the Lean 4 language server. I needed this, so I decided to go ahead and code it myself.

### CODE REVIEW

**Comment by tydeu** [comment]
This can be further generalized to allow the configuration of arbitrary environment variables. However, care still has to be taken with `Path`,  as one generally wants to add to it not replace it completely. I pushed an additional commit that allows demonstrates such an extension.

**Comment by Kha** [comment]
FWIW this change looks good to me even if longer term this should be resolved by https://github.com/leanprover/lean4/issues/419

File: package.json [hunk 1]
```diff
@@ -59,6 +59,24 @@
...
+          "description": "Environment variables to add to the Lean 4 server environment",
+          "additionalProperties": {
+            "type": "string",
+            "description": "environment variable to add"
+          }
+        },
+				"lean4.serverEnvPaths": {
+					"type": "array",
+					"default": [],
+					"description": "Paths to add to the Lean 4 server environment.",
```
Comment (gebner) [inline package.json:74]:
```suggestion
					"description": "Paths to add to the Lean 4 server environment PATH variable.",
```

File: src/leanclient.ts [hunk 2]
```diff
@@ -61,17 +62,23 @@ export class LeanClient implements Disposable {
         if (this.isStarted()) {
             await this.stop()
         }
+        const env = Object.assign({}, process.env, serverEnv());
+        const paths = serverEnvPaths()
+        if (paths.length != 0) {
+            const PathKey = env.Path ? 'Path' : 'PATH'
+            env[PathKey] = paths.join(path.delimiter) + path.delimiter + process.env.Path
```
Comment (gebner) [inline src/leanclient.ts:69]:
```suggestion
            const pathKey = process.platform === 'win32' ? 'Path' : 'PATH'
            env[pathKey] = paths.join(path.delimiter) + path.delimiter + process.env.Path
```

**Review by gebner — APPROVED** [review]
Please also run `npm i` to reformat the `package.json` file.

**Comment by tydeu** [comment]
@gebner Done! 

P.S. I had previously used `npm ci` because I wasn't sure you would want an updated lock file, which `npm i` creates.

**Comment by gebner** [comment]
Thanks.  I don't care too much about the lockfile, but the package.json file should be clean.

**Pull request merged**